src/fasta.cpp

```diff
@@ -157,3 +156,124 @@ bool fasta::check_length(fasta::SequenceList const& sequences, int limit) {
   }
   return result;
 }
+
+fasta::FastaData fasta::get_conf_data(
+    const fasta::FastaData& fasta_data,
+    const f_config::FeatureSettingsMap& f_set, bool gapped) {
+  fasta::FastaData f;
+  f.probabilities = fasta_data.probabilities;
+  f.sequences = fasta_data.sequences;
+  if (!gapped) {
+    f.sequences = remove_gaps(f.sequences);
+  }
+  for (auto feat_it = f_set.begin(); feat_it != f_set.end(); ++feat_it) {
+    assign_feature_by_pattern(f.sequences, feat_it->second.pattern,
+        feat_it->first);
+    for (auto& seq : feat_it->second.positions) {
+      if ((signed)f.sequences.size() > seq.seq_no && seq.seq_no >= 0) {
+        for (auto& pos : seq.positions) {
+          if ((signed)f.sequences[seq.seq_no].residues.size() > pos && pos >= 0) {
+            f.sequences[seq.seq_no].residues[pos].features.push_back(
+                feat_it->first);
+          }
+          else {
+            std::cout << "Warning: feature positions should be in range: 1 - "
+                          << "sequence length, feature " << feat_it->first
+                          << " cannot be annotated at position " << pos
+                          << " in sequence " << seq.seq_no << std::endl;
+          }
+        }
+      }
+      else {
+        std::cout << "Warning: sequence numbers should be in range: 1 - "
```

Comment:
In my opinion this should be fatal (and the same for the one above). A warning may be missed. I don't see any reason why anyone would want to continue if their custom features contain invalid sequence and/or residue position.